File: server/src/main/java/model/member/generation/Talent.java
package model.member.generation;

public class Talent {

    protected String name;
    protected PrimaryAttribute[] attributes;
    protected boolean magicTalent;
    protected boolean weaponTalent;

    public Talent() {
        this.name = "";
        this.attributes = new PrimaryAttribute[0];
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public PrimaryAttribute[] getAttributes() {
        return attributes;
    }

    public void setAttributes(PrimaryAttribute[] attributes) {
        this.attributes = attributes;
    }

    public boolean isMagicTalent() {
        return magicTalent;
    }

    public void setMagicTalent(boolean magicTalent) {
        this.magicTalent = magicTalent;
    }

    public boolean isWeaponTalent() {
        return weaponTalent;
    }

    public void setWeaponTalent(boolean weaponTalent) {
        this.weaponTalent = weaponTalent;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/de/pnp/manager/app/model/Talent.kt
package de.pnp.manager.app.model

import de.pnp.manager.model.character.data.IPrimaryAttribute
import de.pnp.manager.model.other.ITalent

class Talent : ITalent {
    private var name : String = ""
    private var attributes : Array<IPrimaryAttribute> = Array(0){PrimaryAttribute.DUMMY}
    private var magicTalent : Boolean = false
    private var weaponTalent: Boolean = false

    override fun getName(): String {
        return name
    }

    override fun setName(name: String?) {
        this.name = name?:""
    }

    override fun getAttributes(): Array<IPrimaryAttribute> {
        return attributes
    }

    override fun setAttributes(attributes: Array<IPrimaryAttribute>) {
        this.attributes = attributes
    }

    override fun isMagicTalent(): Boolean {
        return this.magicTalent
    }

    override fun setMagicTalent(magicTalent: Boolean) {
        this.magicTalent = magicTalent
    }

    override fun isWeaponTalent(): Boolean {
        return weaponTalent
    }

    override fun setWeaponTalent(weaponTalent: Boolean) {
        this.weaponTalent = weaponTalent
    }
}